nixos/modules/system/activation/activation-script.nix

```diff
@@ -137,8 +140,13 @@ in
 
         mkdir -m 1777 -p /var/tmp
 
+        # Make sure it's really empty
+        chattr -i /var/empty
```

Comment:
This could be `${e2fsprogs}/bin/chattr` to prevent polluting the path with `e2fsprogs`.